### PR TITLE
Docs: dockerless builds, runners, polling, server stored waypoint.hcl, more

### PR DESCRIPTION
**Pull request by mitchellh**
This lays a solid foundation of docs for the aforementioned (subject) features. I think there is a lot of room for improvement but this gets us along.

There are `TODO`s in here on purpose waiting on UI screenshots I'll grab later.

### CODE REVIEW

File: website/content/commands/init.mdx [hunk 1]
```diff
@@ -25,8 +25,8 @@ Usage: `waypoint init [options]`
 
 #### Command Options
 
-- `-from-project=<string>` - Create a new application by fetching the given application from a remote source
-- `-into=<string>` - Where to write the application fetched via -from
+- `-from-project=<string>` - Create a new application by fetching the given application froma remote source or from a local project folder or fileon disk.
```
Comment (briancain) [inline website/content/commands/init.mdx:28]:
```suggestion
- `-from-project=<string>` - Create a new application by fetching the given application from a remote source or from a local project folder or file on disk.
```

Comment (briancain) [reply website/content/commands/init.mdx:28]: Ah, I know this is auto generated, so maybe the fix is made else where :smile:

File: website/content/docs/projects/git.mdx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+```hcl
+runner {
+  enable = true
+
+  data_source "git" {
+    url  = "https://github.com/hashicorp/waypoint-examples.git"
+	path = "docker/node-js"
+  }
+}
```
Comment (briancain) [inline website/content/docs/projects/git.mdx:43]:
```suggestion
  data_source "git" {
    url  = "https://github.com/hashicorp/waypoint-examples.git"
    path = "docker/node-js"
  }
}
```

File: website/content/docs/waypoint-hcl/runner.mdx [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+```hcl
+runner {
+  enable = true
+
+  data_source "git" {
+    url  = "https://github.com/hashicorp/waypoint-examples.git"
+	path = "docker/node-js"
+  }
+}
```
Comment (briancain) [inline website/content/docs/waypoint-hcl/runner.mdx:29]:
```suggestion
  data_source "git" {
    url  = "https://github.com/hashicorp/waypoint-examples.git"
    path = "docker/node-js"
  }
}
```

**Review by briancain — APPROVED** [review]
NIce! :runner: :scroll: Looks good, just found a few typos :smile:

**Review by evanphx — APPROVED** [review]
Looks good!

**Comment by mitchellh** [comment]
Gonna merge to get it on tip but any more feedback happy to address!

**Pull request merged**